sycl/test-e2e/KernelAndProgram/kernel-bundle-merge-options.cpp

```diff
@@ -1,21 +1,24 @@
 // REQUIRES: gpu
-// RUN: %{build} -o %t.out %debug_option
+// RUN: %{build} -o %t.out -g
 // RUN: env SYCL_UR_TRACE=2 %{run} %t.out | FileCheck %s
 // UNSUPPORTED: hip
 
-// Debug option -g is not passed to device code compiler when CL-style driver
-// is used and /DEBUG options is passed.
-// XFAIL: cl_options
-// XFAIL-TRACKER: https://github.com/intel/llvm/issues/16309
+// Rather than %debug_option, we just use -g since it's supported everywhere
+// and simplifies things on the FileCheck side.
+
+// Note that the UR call might be urProgramBuild OR urProgramBuildExp .
+// The same is true for Compile and Link.
+// We want the first match. Don't put parentheses after.
 
 #include "kernel-bundle-merge-options.hpp"
 
 // CHECK: <--- urProgramBuild
 // CHECK-SAME: -g
 
-// TODO: Uncomment when build options are properly passed to compile and link
+// CHECK: <--- urProgramCompile
+// CHECK-SAME: -g
+
+// TODO: Uncomment when build options are properly passed to link
 //       commands for kernel_bundle
-// xCHECK: <--- urProgramCompile(
-// xCHECK-SAME: -g
-// xCHECK: <--- urProgramLink(
+// xCHECK: <--- urProgramLink
```

Comment:
I think there's an ongoing effort to remove this kind of directives (xRUN, xCHECK). @AlexeySachkov can you confirm?

Reply:
There isn't specific effort for `xRUN`/`xCHECK`, but while we are here we should submit a tracker and reference it from the test, or otherwise this `TODO` won't ever be done and we won't even know that we have a bug/missing functionality.